src/client/app/components/unit/CreateUnitModalComponent.tsx

```diff
@@ -69,11 +69,13 @@ export default function CreateUnitModalComponent() {
 	/* Create Unit Validation:
 		Name cannot be blank
 		Sec in Rate must be greater than zero
+		If type of unit is suffix their must be a suffix
 	*/
 	const [validUnit, setValidUnit] = useState(false);
 	useEffect(() => {
-		setValidUnit(state.name !== '' && state.secInRate > 0);
-	}, [state.name, state.secInRate]);
+		setValidUnit(state.name !== '' && state.secInRate > 0 &&
+			(state.typeOfUnit === UnitType.suffix && state.suffix !== ''));
```

Comment:
I think the logic is off here. I believe the correct logic is
```
	setValidUnit(state.name !== '' && state.secInRate > 0 &&
			(state.typeOfUnit !== UnitType.suffix || state.suffix !== ''));
```
I found the current logic did not allow saving if you removed the suffix and then changed the type of unit to non-suffix. Let me know if you agree and, if so, change in both files.

Reply:
Your correct I see the same issue and after using your logic it seems to be working fine I apologize.